components/mpas-ocean/src/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -521,49 +529,71 @@ subroutine ocn_tracer_hmix_Redi_init(domain, err)!{{{
          call mpas_pool_get_dimension(meshPool, 'nEdges', nEdges)
          call mpas_pool_get_array(diagnosticsPool, 'RediKappa', RediKappa)
+         call mpas_pool_get_array(diagnosticsPool, 'RediHorizontalTaper', RediHorizontalTaper)
          call mpas_pool_get_array(forcingPool, 'RediKappaData', RediKappaData)
          call mpas_pool_get_array(meshPool, 'cellsOnEdge', cellsOnEdge)
          call mpas_pool_get_array(meshPool, 'dcEdge', dcEdge)
-         ! initialize Redi kappa array
-         if (config_Redi_closure == 'constant') then
+
+         ! Initialize horizonal taper
+         if (config_Redi_horizontal_taper == 'none' .or. &
+             config_Redi_horizontal_taper == 'RossbyRadius') then
+            ! For 'RossbyRadius', the taper is recomputed at every time step in
+            ! the ocn_GM_compute_Bolus_velocity subroutine
             !$omp parallel
             !$omp do schedule(runtime)
-            do iEdge = 1, nEdges
-               RediKappa(iEdge) = config_Redi_constant_kappa
+            do iEdge = 1,nEdges
+               RediHorizontalTaper(iEdge) = 1.0_RKIND
+            end do
+            !$omp end do
+            !$omp end parallel
+         else if (config_Redi_horizontal_taper == 'ramp') then
+            coef = 1.0_RKIND &
+                    /(config_Redi_horizontal_ramp_max - config_Redi_horizontal_ramp_min)
+            !$omp parallel
+            !$omp do schedule(runtime) 
+            do iEdge=1,nEdges
+               if (dcEdge(iEdge) <= config_Redi_horizontal_ramp_min) then
+                  RediHorizontalTaper(iEdge) = 0.0_RKIND
+               else if (dcEdge(iEdge) >= config_Redi_horizontal_ramp_max) then
+                  RediHorizontalTaper(iEdge) = 1.0_RKIND
+               else
+                  RediHorizontalTaper(iEdge) = coef &
+                      *(dcEdge(iEdge) - config_Redi_horizontal_ramp_min)
+               end if
             end do
             !$omp end do
             !$omp end parallel
-          else if (config_Redi_closure == 'data') then
-            ! read RediKappa in from input
-            call mpas_log_write("config_Redi_closure = 'data'. "// &
-                                "Make sure that the variable RediKappa is read in from an input file.")
          else
-            call mpas_log_write('Invalid choice of config_Redi_closure.', MPAS_LOG_CRIT)
+            call mpas_log_write('Invalid choice of config_Redi_horizontal_taper.', MPAS_LOG_CRIT)
             err = 1
             call mpas_dmpar_finalize(domain%dminfo)
          end if
 
-         ! Add resolution taper
-         if (config_eddying_resolution_taper == 'none') then
-            ! Nothing to do, as we just keep the same assignment as above.
-         else if (config_eddying_resolution_taper == 'ramp') then
+         ! initialize Redi kappa array
+         if (config_Redi_closure == 'constant'.or. &
+             config_Redi_closure == 'equalGM') then
```

Comment:
@mark-petersen I thought 'equalGM' is being moved to a different PR?  It appears here.

Reply:
Thanks, I removed that line.